vendor/bat-native-ads/src/bat/ads/internal/database/tables/creative_new_tab_page_ad_wallpapers_database_table.cc

```diff
@@ -0,0 +1,142 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/database/tables/creative_new_tab_page_ad_wallpapers_database_table.h"
+
+#include <utility>
+
+#include "base/check.h"
+#include "base/strings/stringprintf.h"
+#include "bat/ads/ads_client.h"
+#include "bat/ads/internal/ads_client_helper.h"
+#include "bat/ads/internal/database/database_statement_util.h"
+#include "bat/ads/internal/database/database_table_util.h"
+#include "bat/ads/internal/database/database_util.h"
+
+namespace ads {
+namespace database {
+namespace table {
+
+namespace {
+
+const char kTableName[] = "creative_new_tab_page_ad_wallpapers";
+
+int BindParameters(mojom::DBCommand* command,
+                   const CreativeNewTabPageAdList& creative_ads) {
+  DCHECK(command);
+
+  int count = 0;
+  int index = 0;
+
+  for (const auto& creative_ad : creative_ads) {
+    for (const auto& wallpaper : creative_ad.wallpapers) {
+      BindString(command, index++, creative_ad.creative_instance_id);
+      BindString(command, index++, wallpaper.image_url);
+      BindInt(command, index++, wallpaper.focal_point.x);
+      BindInt(command, index++, wallpaper.focal_point.y);
+
+      count++;
```

Comment:
Do we really need to calculate `count`? Seems like it's just `creative_ad.wallpapers.size()` unless I'm missing something.

Reply:
This is a total count of all wallpapers for each creative ad

Reply:
Ah yes, missed the other loop!

Reply:
I've moved the counter outside the loop though as a good catch